code/modules/mob/living/living.dm

```diff
@@ -41,7 +41,7 @@
 
 
 /mob/living/proc/handle_staminaloss()
-	if(world.time < last_staminaloss_dmg + 3 SECONDS || (m_intent == MOVE_INTENT_RUN && world.time < last_move_intent + 1 SECONDS))
+	if(world.time < last_staminaloss_dmg + 3 SECONDS)
```

Comment:
Would now be a good time to turn this into a `cooldown`?

Reply:
This would have to be a stoppable timer, not a regular cooldown, which is more expensive.
I don't see much gain in it, to be fair, this method is by far the cheapest.